extra-kotlin/src/main/kotlin/net/kyori/adventure/extra/kotlin/KeyExtensions.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * This file is part of adventure, licensed under the MIT License.
+ *
+ * Copyright (c) 2017-2021 KyoriPowered
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package net.kyori.adventure.extra.kotlin
+
+import net.kyori.adventure.key.Key
+
+/**
+ * The namespace of this key.
+ *
+ * Allows for destructuring into (namespace, key)
+ *
+ * @return the namespace
+ * @since 4.8.0
+ */
+public fun Key.component1(): String = namespace()
```

Comment:
Unsure about this sort of destructuring -- i'd lean towards not including this, unless someone else comes out strongly in favor.

Reply:
I think it's kinda similar to the way you'd use `split` on the pure string version and destructure that into the namespace and the key, e.g. `val (namespace, key) = stringKey.split(":")`

Reply:
:+1: with what @zml2008 said.